src/db.c

```diff
@@ -606,6 +607,10 @@ void setKeyByLink(client *c, redisDb *db, robj *key, robj **valref, int flags, d
         signalModifiedKey(c,db,key);
 }
 
+static int randomKeyShouldSkipDictIndex(int didx) {
+    return server.cluster_enabled && !asmSlotAllowsExpiryOrEviction(didx);
```

Comment:
now I wonder how are we going to do this for destination side replica? Replica is not aware of the migration.

Reply:
oh, i missed this point

Reply:
I missed it as well :D 

Replica may check if it is the owner of the slot but as replica does not know when migration started, we have to do the check all the time. Then, I wonder if there will be performance cost somewhere.